### PR TITLE
HBASE-28767 Simplify backup bulk-loading code

### PR DESCRIPTION
**Pull request by DieterDP-ng**
No functional changes.

Currently split into 2 commits as to help the reviewer with seeing what was changed. This can be squashed when merging.

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 37s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  |  Patch does not have any anti-patterns.  |
|||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m  3s |  |  master passed  |
| +1 :green_heart: |  compile  |   0m 28s |  |  master passed  |
| +1 :green_heart: |  checkstyle  |   0m 10s |  |  master passed  |
| +1 :green_heart: |  spotbugs  |   0m 28s |  |  master passed  |
| +1 :green_heart: |  spotless  |   0m 43s |  |  branch has no errors when running spotless:check.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 55s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 27s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 27s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| -0 :warning: |  checkstyle  |   0m  9s | [/results-checkstyle-hbase-backup.txt](https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6134/1/artifact/yetus-general-check/output/results-checkstyle-hbase-backup.txt) |  hbase-backup: The patch generated 5 new + 0 unchanged - 0 fixed = 5 total (was 0)  |
| +1 :green_heart: |  spotbugs  |   0m 34s |  |  the patch passed  |
| +1 :green_heart: |  hadoopcheck  |  11m  1s |  |  Patch does not cause any errors with Hadoop 3.3.6 3.4.0.  |
| +1 :green_heart: |  spotless  |   0m 42s |  |  patch has no errors when running spotless:check.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 10s |  |  The patch does not generate ASF License warnings.  |
|  |   |  28m  9s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6134/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6134 |
| Optional Tests | dupname asflicense javac spotbugs checkstyle codespell detsecrets compile hadoopcheck hbaseanti spotless |
| uname | Linux 59a1e8036050 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 3809779a94f45e8080306229bd8d6bed35e573e0 |
| Default Java | Eclipse Adoptium-17.0.11+9 |
| Max. process+thread count | 85 (vs. ulimit of 30000) |
| modules | C: hbase-backup U: hbase-backup |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6134/1/console |
| versions | git=2.34.1 maven=3.9.8 spotbugs=4.7.3 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 38s |  |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --author-ignore-list --blanks-eol-ignore-file --blanks-tabs-ignore-file --quick-hadoopcheck  |
|||| _ Prechecks _ |
|||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m  8s |  |  master passed  |
| +1 :green_heart: |  compile  |   0m 22s |  |  master passed  |
| +1 :green_heart: |  javadoc  |   0m 16s |  |  master passed  |
| +1 :green_heart: |  shadedjars  |   5m 28s |  |  branch has no errors when building our shaded downstream artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m  5s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 20s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 20s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   0m 15s |  |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   5m 21s |  |  patch has no errors when building our shaded downstream artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |  10m 39s |  |  hbase-backup in the patch passed.  |
|  |   |  30m 27s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.46 ServerAPI=1.46 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6134/1/artifact/yetus-jdk17-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6134 |
| Optional Tests | javac javadoc unit compile shadedjars |
| uname | Linux e891af83a1eb 5.4.0-177-generic #197-Ubuntu SMP Thu Mar 28 22:45:47 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 3809779a94f45e8080306229bd8d6bed35e573e0 |
| Default Java | Eclipse Adoptium-17.0.11+9 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6134/1/testReport/ |
| Max. process+thread count | 3455 (vs. ulimit of 30000) |
| modules | C: hbase-backup U: hbase-backup |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6134/1/console |
| versions | git=2.34.1 maven=3.9.8 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

File: hbase-backup/src/main/java/org/apache/hadoop/hbase/backup/impl/BackupSystemTable.java [hunk 1]
```diff
@@ -792,20 +755,19 @@ public List<BackupInfo> getBackupHistory(int n, BackupInfo.Filter... filters) th
...
 
-  /*
-   * Retrieve TableName's for completed backup of given type
-   * @param type backup type
-   * @return List of table names
+  /**
+   * Retrieve all table names that are part of any known backup
    */
-  public List<TableName> getTablesForBackupType(BackupType type) throws IOException {
+  public Set<TableName> getTablesForBackupType() throws IOException {
```
Comment (DieterDP-ng) [inline hbase-backup/src/main/java/org/apache/hadoop/hbase/backup/impl/BackupSystemTable.java:743]: This method should probably be renamed.

File: hbase-backup/src/main/java/org/apache/hadoop/hbase/backup/impl/IncrementalTableBackupClient.java [hunk 2]
```diff
@@ -175,24 +159,20 @@ protected Map<byte[], List<Path>>[] handleBulkLoad(List<TableName> sTableList)
             if (fs.exists(p)) {
               if (LOG.isTraceEnabled()) {
                 LOG.trace("found bulk hfile " + file + " in " + famDir + " for " + tblName);
```
Comment (ndimiduk) [inline hbase-backup/src/main/java/org/apache/hadoop/hbase/backup/impl/IncrementalTableBackupClient.java]: nit: clean up these log lines with `{}` as well.

File: hbase-backup/src/main/java/org/apache/hadoop/hbase/backup/impl/BackupManager.java [hunk 3]
```diff
@@ -356,8 +355,7 @@ public HashMap<String, Long> readRegionServerLastLogRollResult() throws IOExcept
     return systemTable.readRegionServerLastLogRollResult(backupInfo.getBackupRootDir());
   }
 
-  public Pair<Map<TableName, Map<String, Map<String, List<Pair<String, Boolean>>>>>, List<byte[]>>
-    readBulkloadRows(List<TableName> tableList) throws IOException {
+  public List<BulkLoad> readBulkloadRows(List<TableName> tableList) throws IOException {
```
Comment (ndimiduk) [inline hbase-backup/src/main/java/org/apache/hadoop/hbase/backup/impl/BackupManager.java:358]: a type!! so much better.

**Review by ndimiduk — CHANGES_REQUESTED** [review]
Much better than it was. thank you @DieterDP-ng

File: hbase-backup/src/main/java/org/apache/hadoop/hbase/backup/impl/BackupSystemTable.java (hunk 1 above)
Comment (rmdmattingly) [reply hbase-backup/src/main/java/org/apache/hadoop/hbase/backup/impl/BackupSystemTable.java:743]: Yeah, getFullyBackedUpTables or getTablesWithFullBackups would be improve the clarity

File: hbase-backup/src/main/java/org/apache/hadoop/hbase/backup/impl/BackupSystemTable.java [hunk 4]
```diff
@@ -400,43 +395,22 @@ public void deleteBackupInfo(String backupId) throws IOException {
...
+    Map<byte[], List<Path>> cfToHfilePath) throws IOException {
     if (LOG.isDebugEnabled()) {
-      LOG.debug("write bulk load descriptor to backup " + tabName + " with " + finalPaths.size()
-        + " entries");
+      LOG.debug("Writing bulk load descriptor to backup {} with {} entries", tabName,
+        cfToHfilePath.size());
     }
     try (Table table = connection.getTable(bulkLoadTableName)) {
-      List<Put> puts = BackupSystemTable.createPutForCommittedBulkload(tabName, region, finalPaths);
+      List<Put> puts = BackupSystemTable.createPutForBulkLoad(tabName, region, cfToHfilePath);
```
Comment (rmdmattingly) [inline hbase-backup/src/main/java/org/apache/hadoop/hbase/backup/impl/BackupSystemTable.java:411]: While we're here, it's probably worth using a BufferedMutator like we did in https://github.com/apache/hbase/pull/6067

File: hbase-backup/src/main/java/org/apache/hadoop/hbase/backup/impl/BackupSystemTable.java [hunk 5]
```diff
@@ -400,43 +395,22 @@ public void deleteBackupInfo(String backupId) throws IOException {
...
-    if (LOG.isDebugEnabled()) {
-      LOG.debug(
-        "write bulk load descriptor to backup " + tabName + " with " + pairs.size() + " entries");
-    }
-    try (Table table = connection.getTable(bulkLoadTableName)) {
-      List<Put> puts =
-        BackupSystemTable.createPutForPreparedBulkload(tabName, region, family, pairs);
-      table.put(puts);
-      LOG.debug("written " + puts.size() + " rows for bulk load of " + tabName);
+      LOG.debug("Written {} rows for bulk load of {}", puts.size(), tabName);
```
Comment (rmdmattingly) [inline hbase-backup/src/main/java/org/apache/hadoop/hbase/backup/impl/BackupSystemTable.java:413]: while we're here, can we change `tabName` to `tableName`?

File: hbase-backup/src/main/java/org/apache/hadoop/hbase/backup/impl/BackupSystemTable.java [hunk 6]
```diff
@@ -1579,30 +1521,8 @@ public static void deleteSnapshot(Connection conn) throws IOException {
...
-      put.addColumn(BackupSystemTable.META_FAMILY, PATH_COL, Bytes.toBytes(file));
-      put.addColumn(BackupSystemTable.META_FAMILY, STATE_COL, BL_PREPARE);
-      puts.add(put);
-      LOG.debug("writing raw bulk path " + file + " for " + table + " " + Bytes.toString(region));
-    }
-    return puts;
-  }
-
   public static List<Delete> createDeleteForOrigBulkLoad(List<TableName> lst) {
+    // Unused???
```
Comment (rmdmattingly) [inline hbase-backup/src/main/java/org/apache/hadoop/hbase/backup/impl/BackupSystemTable.java:1525]: should we act on this comment now?

Comment (DieterDP-ng) [reply hbase-backup/src/main/java/org/apache/hadoop/hbase/backup/impl/BackupSystemTable.java:1525]: Ill deal with it in HBASE-29003.

**Comment by DieterDP-ng** [comment]
Apologies for letting this gather dust. Review comments have been addressed.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m  0s |  |  Docker mode activated.  |
| -1 :x: |  docker  |   0m  4s |  |  Docker failed to build run-specific yetus/hbase:tp-18400}.  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| GITHUB PR | https://github.com/apache/hbase/pull/6134 |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6134/2/console |
| versions | git=2.17.1 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 30s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  |  Patch does not have any anti-patterns.  |
|||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m  4s |  |  master passed  |
| +1 :green_heart: |  compile  |   0m 29s |  |  master passed  |
| +1 :green_heart: |  checkstyle  |   0m 10s |  |  master passed  |
| +1 :green_heart: |  spotbugs  |   0m 30s |  |  master passed  |
| +1 :green_heart: |  spotless  |   0m 43s |  |  branch has no errors when running spotless:check.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 50s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 28s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 28s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| -0 :warning: |  checkstyle  |   0m  7s | [/buildtool-patch-checkstyle-hbase-backup.txt](https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6134/2/artifact/yetus-general-check/output/buildtool-patch-checkstyle-hbase-backup.txt) |  The patch fails to run checkstyle in hbase-backup  |
| +1 :green_heart: |  spotbugs  |   0m 33s |  |  the patch passed  |
| +1 :green_heart: |  hadoopcheck  |  10m 53s |  |  Patch does not cause any errors with Hadoop 3.3.6 3.4.0.  |
| +1 :green_heart: |  spotless  |   0m 42s |  |  patch has no errors when running spotless:check.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 10s |  |  The patch does not generate ASF License warnings.  |
|  |   |  28m  7s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6134/2/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6134 |
| Optional Tests | dupname asflicense javac spotbugs checkstyle codespell detsecrets compile hadoopcheck hbaseanti spotless |
| uname | Linux 3dc76d07d0ce 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 5932ae3f2088948dd24eca6c831e2ab90b8da093 |
| Default Java | Eclipse Adoptium-17.0.11+9 |
| Max. process+thread count | 83 (vs. ulimit of 30000) |
| modules | C: hbase-backup U: hbase-backup |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6134/2/console |
| versions | git=2.34.1 maven=3.9.8 spotbugs=4.7.3 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by DieterDP-ng** [comment]
> 💔 **-1 overall**
> Vote 	Subsystem 	Runtime 	Logfile 	Comment
> +0 🆗 	reexec 	0m 0s 		Docker mode activated.
> -1 ❌ 	docker 	0m 4s 		Docker failed to build run-specific yetus/hbase:tp-18400}.
> Subsystem 	Report/Notes
> GITHUB PR 	#6134
> Console output 	https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6134/2/console
> versions 	git=2.17.1
> Powered by 	Apache Yetus 0.15.0 https://yetus.apache.org
> 
> This message was automatically generated.

This looks a failure unrelated to the PR to me. Let me know if this is not the case.

**Review by ndimiduk — APPROVED** [review]
Thanks for addressing my nits. This is looking good by me. I've initiated another PR build since the last one had some infra-level failure.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 28s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  |  Patch does not have any anti-patterns.  |
|||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m  5s |  |  master passed  |
| +1 :green_heart: |  compile  |   0m 29s |  |  master passed  |
| +1 :green_heart: |  checkstyle  |   0m 10s |  |  master passed  |
| +1 :green_heart: |  spotbugs  |   0m 32s |  |  master passed  |
| +1 :green_heart: |  spotless  |   0m 44s |  |  branch has no errors when running spotless:check.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 56s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 30s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 30s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| -0 :warning: |  checkstyle  |   0m  7s | [/buildtool-patch-checkstyle-hbase-backup.txt](https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6134/3/artifact/yetus-general-check/output/buildtool-patch-checkstyle-hbase-backup.txt) |  The patch fails to run checkstyle in hbase-backup  |
| +1 :green_heart: |  spotbugs  |   0m 37s |  |  the patch passed  |
| +1 :green_heart: |  hadoopcheck  |  10m 59s |  |  Patch does not cause any errors with Hadoop 3.3.6 3.4.0.  |
| +1 :green_heart: |  spotless  |   0m 43s |  |  patch has no errors when running spotless:check.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 10s |  |  The patch does not generate ASF License warnings.  |
|  |   |  28m 15s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6134/3/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6134 |
| Optional Tests | dupname asflicense javac spotbugs checkstyle codespell detsecrets compile hadoopcheck hbaseanti spotless |
| uname | Linux 8c011f88d156 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 5932ae3f2088948dd24eca6c831e2ab90b8da093 |
| Default Java | Eclipse Adoptium-17.0.11+9 |
| Max. process+thread count | 83 (vs. ulimit of 30000) |
| modules | C: hbase-backup U: hbase-backup |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6134/3/console |
| versions | git=2.34.1 maven=3.9.8 spotbugs=4.7.3 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 20s |  |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --author-ignore-list --blanks-eol-ignore-file --blanks-tabs-ignore-file --quick-hadoopcheck  |
|||| _ Prechecks _ |
|||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 13s |  |  master passed  |
| +1 :green_heart: |  compile  |   0m 20s |  |  master passed  |
| +1 :green_heart: |  javadoc  |   0m 15s |  |  master passed  |
| +1 :green_heart: |  shadedjars  |   4m 54s |  |  branch has no errors when building our shaded downstream artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 39s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 20s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 20s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   0m 14s |  |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   4m 51s |  |  patch has no errors when building our shaded downstream artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |  15m 35s |  |  hbase-backup in the patch passed.  |
|  |   |  33m 27s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.47 ServerAPI=1.47 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6134/3/artifact/yetus-jdk17-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6134 |
| Optional Tests | javac javadoc unit compile shadedjars |
| uname | Linux f67eedb73dd9 5.4.0-200-generic #220-Ubuntu SMP Fri Sep 27 13:19:16 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 5932ae3f2088948dd24eca6c831e2ab90b8da093 |
| Default Java | Eclipse Adoptium-17.0.11+9 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6134/3/testReport/ |
| Max. process+thread count | 2972 (vs. ulimit of 30000) |
| modules | C: hbase-backup U: hbase-backup |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6134/3/console |
| versions | git=2.34.1 maven=3.9.8 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

**Pull request merged**